### DISCUSSION TITLE
Ability to ignore previous issues based on code signature

### DISCUSSION THREAD

**Discussion by estahn** [astral-sh/ruff#14619] [Ideas]
# Problem
We're introducing ruff to our codebase and ideally would like to roll out stricter rulesets without the need to fix 20k+ issues.

# Possible Solution
Create a "database" containing all findings to be ignored. The findings could be a hash-map consisting of hash of line of code and the reported error code. Upon the next run ruff could use this to exclude those findings.

Notes:
* Don't use `column`/`row` as this will eventually change over time
* The database can be updated over time if needed
* If the signature changes the code has changed and issues should be addressed
* Possibly use something like MD5 of code and filename

**Example:**

Finding
```
55 | def test_do_FooBar(api_client, authenticated_user, mock_pynamodb, mock_eventbridge_client):
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ N802
```

Create findings to be ignored:
```bash
ruff check --findings-out=ruff-ignore.txt
```

ruff-ignore.txt
```
<md5 of line 55>:<error code>
9156082527e60149f4e08c7601f00539:N802
```

Thoughts?

**Comment by zanieb** [comment]
Previous discussion at https://github.com/astral-sh/ruff/issues/1149

**Reply by estahn** [reply]
@zanieb Thanks, I couldn't find a discussion around that topic. Possibly used the incorrect terminology.
